### ISSUE TITLE
Why use exponential distribution to implement Gumbel softmax?

### ISSUE THREAD

**Issue by pingguokiller** [D-X-Y/AutoDL-Projects#89]
In function select2withP of AutoDL-Projects\lib\models\shape_searchs\SoftSelect.py, the implementation of Gumbel softmax is below:

gumbels = -torch.empty_like(logits).exponential_().log()

I don't understand why to use exponential distribution to implement Gumbel softmax?

**Comment by D-X-Y** [comment]
I follow the implementation of PyTorch at here: https://pytorch.org/docs/stable/_modules/torch/nn/functional.html#gumbel_softmax
